cub/cub/agent/agent_select_if.cuh

```diff
@@ -304,11 +310,12 @@ struct AgentSelectIf
 
   _TempStorage& temp_storage; ///< Reference to temp_storage
   WrappedInputIteratorT d_in; ///< Input items
-  OutputIteratorWrapperT d_selected_out; ///< Unique output items
+  OutputIteratorWrapperT d_selected_out; ///< Output iterator for the selected items
   WrappedFlagsInputIteratorT d_flags_in; ///< Input selection flags (if applicable)
   InequalityWrapper<EqualityOpT> inequality_op; ///< T inequality operator
   SelectOpT select_op; ///< Selection operator
   OffsetT num_items; ///< Total number of input items
+  const StreamingContextT& streaming_context; ///< Context for the current partition
```

Comment:
It's really that the compiler produces "poor" codegen without it being a constant reference. I experienced it throughout the process of mitigating performance regressions but decided to give it another try today, on the current state of the PR. It turns out that it **still degrades performance** - in the worst case by 17% in the streaming case. So, I'm afraid we have to keep it that way.

Reply:
Could you please add a comment on that so that this is not accidentally refactored later?

Reply:
I've added the following comment:
> // Note: This is a const reference because we have seen double-digit percentage perf regressions otherwise

Sorry, I had missed that suggestion from your earlier comment.